digital_comms/fixed_network/interventions.py

```diff
@@ -26,51 +26,36 @@ def get_all_assets_ranked(system, ranking_variable, asset_variable, technology,
         preference.
 
     """
-    if asset_variable == 'exchange':
-        system_level = system._exchanges
-    elif asset_variable == 'cabinet':
-        system_level = system._cabinets
-    elif asset_variable == 'distribution':
-        system_level = system._distributions
-    else:
-        raise ValueError('Did not recognise asset_variable')
-
     if ranking_variable == 'rollout_benefits':
```

Comment:
I think you can generalise this like so:
```python
    assets = sorted(
        (asset for asset in system_assets
         if getattr(asset, technology) != asset.total_prems),
         key=lambda item: getattr(item, ranking_variable)[technology], reverse=reverse_value)
```
which means you don't need all those if-then-else statements.